### PR TITLE
Handle discussion reply:

### PR DESCRIPTION
**Pull request by lavanyachandramouli**
1. Update Title value with parent discussion title
2. Add source url parameter in display url. On click of Ok, SP navigates
back to parent discussion thread instead of list.

### CODE REVIEW

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 1]
```diff
@@ -3044,6 +3046,20 @@ private void getListItemDocContent(Request request, Response response,
           log.exiting("SiteAdaptor", "getListItemDocContent");
           return;
         }
+        if ("Message".equals(contentType)) {       
+        // discussion replies will be of content type "Message".
+        // There is no title field for content type: Message.
+        // So update the title value as discussiontitle
+          title = row.getAttribute("ows_DiscussionTitle");
+          String sourceURL = row.getAttribute("ows_FileDirRef").split(";#")[1];
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: s/sourceURL/sourceUrl/. Use OWS_FILEDIRREF_ATTRIBUTE. Add 2 as second argument to split (mainly for consistency).

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 2]
```diff
@@ -3044,6 +3046,20 @@ private void getListItemDocContent(Request request, Response response,
           log.exiting("SiteAdaptor", "getListItemDocContent");
           return;
         }
+        if ("Message".equals(contentType)) {
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: This should go before the check of canRespondWithNoContent just above this.

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 3]
```diff
@@ -3044,6 +3046,20 @@ private void getListItemDocContent(Request request, Response response,
           log.exiting("SiteAdaptor", "getListItemDocContent");
           return;
         }
+        if ("Message".equals(contentType)) {       
+        // discussion replies will be of content type "Message".
+        // There is no title field for content type: Message.
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: I think briefer is OK here, just a one line comment:

// There is no title field for discussion replies, use DiscussionTitle.

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 4]
```diff
@@ -196,6 +196,7 @@
    * for more information about content type IDs.
    */
   private static final String OWS_CONTENTTYPEID_ATTRIBUTE = "ows_ContentTypeId";
+  private static final String OWS_CONTENTTYPE_ATTRIBUTE = "ows_ContentType";
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Add javadoc.

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 5]
```diff
@@ -3006,6 +3007,7 @@ private void getListItemDocContent(Request request, Response response,
         return;
       }
       String contentTypeId = row.getAttribute(OWS_CONTENTTYPEID_ATTRIBUTE);
+      String contentType = row.getAttribute(OWS_CONTENTTYPE_ATTRIBUTE);
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Move this to just before the if, at least, maybe inline it.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java [hunk 6]
```diff
@@ -2679,7 +2679,162 @@ public void testGetDocContentListItem() throws Exception {
...
+        + "<tr><td>Created</td><td>2012-05-01T22:14:06Z</td></tr>"
+        + "<tr><td>Created Date</td><td>2012-05-01T22:14:06Z</td></tr>"
+        + "<tr><td>DiscussionTitle</td><td>Discussion Subject</td></tr>"
+        + "<tr><td>Editor</td><td>System Account</td></tr>"
+        + "<tr><td>EncodedAbsUrl</td>"
+        +   "<td>http://localhost:1/sites/SiteCollection/Lists/Custom%20List/"
+        +   "Test%20Folder/2_.000</td></tr>"
+        + "<tr><td>FSObjType</td><td>0</td></tr>"
+        + "<tr><td>FileDirRef</td>"
+        + "<td>sites/SiteCollection/Lists/Custom List/Test Folder</td></tr>"
```
Comment (wiarlawd) [inline test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Indent like the other second columns?

**Review by wiarlawd — CHANGES_REQUESTED** [review]

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java (hunk 4 above)
Comment (lavanyachandramouli) [reply src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Done

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java (hunk 5 above)
Comment (lavanyachandramouli) [reply src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Done

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java (hunk 2 above)
Comment (lavanyachandramouli) [reply src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Acknowledge

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java (hunk 3 above)
Comment (lavanyachandramouli) [reply src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Done

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java (hunk 1 above)
Comment (lavanyachandramouli) [reply src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Done

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java (hunk 6 above)
Comment (lavanyachandramouli) [reply test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Done

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 7]
```diff
@@ -3036,14 +3042,30 @@ private void getListItemDocContent(Request request, Response response,
         }
         metadataLength += addMetadata(
             response, METADATA_OBJECT_TYPE, ObjectType.LIST_ITEM.value());
+        String contentType = row.getAttribute(OWS_CONTENTTYPE_ATTRIBUTE);
+        if ("Message".equals(contentType)) {       
+        // discussion replies will be of content type "Message".
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Comment wasn't changed. Replace with a one-liner and indent two spaces:

// There is no title field for discussion replies, use DiscussionTitle.

**Review by wiarlawd — APPROVED** [review]
LGTM++ with a nit.

**Comment by wiarlawd** [comment]
Oh, and update the commit message a bit. The colon is jarring, because the first line is used stand-alone. See if there's a longer title that is more clear (up to 58 characters), or just change colon to period.

**Pull request merged**